corehq/blobs/migrate.py

```diff
@@ -356,6 +356,9 @@ def get_type_code(doc):
         self.get_type_code = get_type_code
 
     def migrate(self, filename=None, reset=False, max_retry=2, chunk_size=100, **kw):
+        if kw.get('date_range'):
+            start, end = kw.get('date_range')
+            self.iteration_key = '{}-{}-{}'.format(self.iteration_key, start, end)
         doc_provider = self.get_document_provider()
```

Comment:
what about passing the key words to `get_document_provider` which is already overridden in `BackendMigrator` and you can be the change to the iteration key there.

I think it's better in `BackendMigrator` since that's the class which supports the date range. Also putting it in `get_document_provider` makes sense to me since that's the only place where the `iteration_key` is used.

It might even be worth making `iteration_key` a method `get_iteration_key`:
```python
class Migrator(object):
    def get_iteration_key(self, **kw):
        doc_types_map = doc_type_tuples_to_dict(self.doc_types)
        sorted_types = sorted(doc_types_map)
        return "{}-blob-migration/{}".format(self.slug, " ".join(sorted_types))

class BackendMigrator(Migrator):
    def get_iteration_key(self, date_range=None, **kw):
        iteration_key = super(BackendMigrator, self).get_iteration_key(**kw)
        if date_range:
            iteration_key = '{}-{}-{}'.format(iteration_key, *date_range)
        return iteration_key
```

Reply:
Ha, makes sense to move it to `get_document_provider`, updated. Thanks!